datastore/transaction_test.go

```diff
@@ -88,3 +88,398 @@ func TestNewTransaction(t *testing.T) {
 		}
 	}
 }
+
```

Comment:
nit: Other tests to consider if necessary:
1. Make sure readTime is passed along for read only transactions
2. Make sure prevId is passed along and used for read/write transactions

Reply:
1. Tested here: https://github.com/googleapis/google-cloud-go/pull/8972/files#diff-8a638c1c614640f81c811cc62b5e9bacbb8b17493e158922ccb9ee004d4ffe70R70-R80
2. prevId is not being used anywhere in Go client library.